lib/js/_validate/_check-package-structure.js

```diff
@@ -116,34 +134,37 @@ function isFileType(item) {
 
 /**
  * Filter glob results
- * - Remove directory string
  * - Remove paths matching .gitignore'd files
  * - Remove dotfiles
  * @private
  * @function getFilteredResults
- * @param {String} directory path to files
  * @param {Array} files file list to filter
  * @return {Array}
  */
-function getFilteredResults(directory, files) {
+function getFilteredResults(files) {
+	// Generate array of glob patterns from .gitignore
 	const globs = gitParse(gitignorePath);
-	const pattern = (globs.length > 1) ? `{${globs.join(',')}}` : globs[0];
-	files = files.map(pathAndFile => pathAndFile.replace(`${directory}/`, ''));
-
-	const exclude = files.filter(
-		minimatch.filter(pattern,
-			{
-				matchBase: true,
-				dot: true
-			}
-		)
-	);
 
-	// https://github.com/regexhq/dotfile-regex/blob/master/index.js
-	const dotfileRegex = /(?:^|[\\\/])(\.(?!\.)[^\\\/]+)$/; // eslint-disable-line no-useless-escape
+	// Generate a pattern matching string from parsed .gitignore
+	const pattern = (globs.length > 1) ? `{${globs.join()}}` : globs[0];
```

Comment:
const pattern = \`{${globs.join()}}\`;
?

Reply:
that is how a pattern is represented in minimatch, as a string of the form `{matcha,matchb,matchc}`

Reply:
i'll add an example to the comment

Reply:
Ahhh, I was thinking the conditional was unnecessary but I think i get it now. You are saying the pattern cannot be `{foo}` i.e. if one entry, it must be a string?

Reply:
yes!